xbmc/cores/VideoRenderers/WinRenderer.cpp

```diff
@@ -1354,9 +1380,120 @@ void YUVBuffer::Clear()
 
 bool YUVBuffer::IsReadyToRender()
 {
+  return !m_locked && !m_pendingCopy;
+}
+
+bool YUVBuffer::CopyFromDXVA(ID3D11VideoDecoderOutputView* pView)
+{
+  if (!pView)
+    return false;
+
+  HRESULT hr = S_OK;
+  D3D11_VIDEO_DECODER_OUTPUT_VIEW_DESC vpivd;
+  pView->GetDesc(&vpivd);
+  ID3D11Resource* resource = nullptr;
+  pView->GetResource(&resource);
+
+  if (!m_staging)
+  {
+    // create staging texture
+    ID3D11Texture2D* surface = nullptr;
+    hr = resource->QueryInterface(__uuidof(ID3D11Texture2D), reinterpret_cast<void**>(&surface));
+    if (SUCCEEDED(hr))
+    {
+      D3D11_TEXTURE2D_DESC tDesc;
+      surface->GetDesc(&tDesc);
+      SAFE_RELEASE(surface);
+
+      CD3D11_TEXTURE2D_DESC sDesc(tDesc);
+      sDesc.ArraySize = 1;
+      sDesc.Usage = D3D11_USAGE_STAGING;
+      sDesc.CPUAccessFlags = D3D11_CPU_ACCESS_READ;
+      sDesc.BindFlags = 0;
+
+      hr = g_Windowing.Get3D11Device()->CreateTexture2D(&sDesc, nullptr, &m_staging);
+      if (SUCCEEDED(hr))
+        m_sDesc = sDesc;
+    }
+  }
+
+  if (m_staging)
+  {
+    ID3D11DeviceContext* pContext = g_Windowing.GetImmediateContext();
+    // copy content from decoder texture to temporary texture.
+    pContext->CopySubresourceRegion(m_staging,
+                                    D3D11CalcSubresource(0, 0, 1),
+                                    0, 0, 0,
+                                    resource,
+                                    D3D11CalcSubresource(0, vpivd.Texture2D.ArraySlice, 1),
+                                    nullptr);
+    m_pendingCopy = true;
+  }
+  SAFE_RELEASE(resource);
+
+  // CopySubresourceRegion is asynchronous operation, let's get d3d11 runtime to copy 
+  // texture in background and then we will do actual copying before rendering
+
+  return SUCCEEDED(hr);
+}
+
+void YUVBuffer::PerformCopy()
```

Comment:
I have no good reason to move this out from decoder thread also because it's used only if user defines <forcedxvarender>false</forcedxvarender> in as.xml. So I think it will be a good compromise to left it as is is and just update to using sse4.